src/packagedcode/npm.py

```diff
@@ -374,22 +431,16 @@ def get_licensing(license_object):
         or
         "licenses": ["MIT"],
         """
-        lics = []
         for lic in license_object:
             if not lic:
                 continue
             if isinstance(lic, string_types):
-                lics.append(lic)
+                declared_licenses.append({'type': lic})
             elif isinstance(lic, dict):
-                lics_val = '\n'.join(v for v in lic.values() if v)
-                lics.append(lics_val)
+                declared_licenses.append(lic)
             else:
-                lics.append(repr(lic))
-        declared_license = u'\n'.join(lics)
-
-    else:
-        declared_license = repr(license_object)
-    return declared_license
+                declared_licenses.append({'type': repr(lic)})
```

Comment:
Same here: I think I get now why you are using a fake type... but the whole purpose of the declared license is to keep the original structure. So by introducing a fake "type" attribute in a mapping here and above you are not faithfully carrying over the license data structure. Furthermore, "type" is the legacy and obsolete way to document npm licenses... so ceating fake ones is not a solution instead we should have this a list (like you have now) with items being either:
- a string, for the cases where we have a single string in `license` or a legacy list of strings in `licenses`
- a mapping as it was e.g. the type/url map

And that's the job of license normalization to normalize this. Normalization should be done before when crafting the declared license field.